cmd/analyze/main.go

```diff
@@ -46,6 +46,14 @@ var (
 		"list of analysis modes to run, separated by commas. Use -list-modes to see available options")
 )
 
+type usageError struct {
+	error
+}
+
+func usagef(format string, args ...any) error {
+	return usageError{fmt.Errorf(format, args...)}
+}
```

Comment:
nit: Is this the custom type necessary if you are not using it to discriminate a exit code?

e.g.

```
func usagef(...) error {
  return fmt.Errorf(...)
}
```

Would suffice in your current implementation.